ios/CodePush/CodePushUpdateUtils.m

```diff
@@ -256,7 +280,88 @@ + (BOOL)verifyHashForDiffUpdate:(NSString *)finalUpdateFolder
         return NO;
     }
     
+    CPLog(@"Expected hash: %@, actual hash: %@", expectedHash, updateContentsManifestHash);
+    
     return [updateContentsManifestHash isEqualToString:expectedHash];
 }
 
-@end
\ No newline at end of file
++ (NSString *)cleanPublicKey:(NSString *)publicKeyString
+{
+    publicKeyString = [publicKeyString stringByReplacingOccurrencesOfString:@"-----BEGIN PUBLIC KEY-----\n"
+                                                                 withString:@""];
+    publicKeyString = [publicKeyString stringByReplacingOccurrencesOfString:@"-----END PUBLIC KEY-----"
+                                                                 withString:@""];
+    publicKeyString = [publicKeyString stringByReplacingOccurrencesOfString:@"\n"
+                                                                 withString:@""];
+
+    return publicKeyString;
+}
+
++ (NSString *)getSignatureFor:(NSString *)folderPath
+                        error:(NSError **)error
+{
+    NSString *signatureFilePath = [NSString stringWithFormat:@"%@/%@/%@", folderPath, ManifestFolderPrefix, BundleJWTFile];
+    if ([[NSFileManager defaultManager] fileExistsAtPath:signatureFilePath]) {
+        return [NSString stringWithContentsOfFile:signatureFilePath encoding:NSUTF8StringEncoding error:error];
+    }else{
+        *error = [CodePushErrorUtils errorWithMessage:[NSString stringWithFormat: @"Cannot find signature at %@", signatureFilePath]];
+        return nil;
+    }
+    return nil;
+}
+
++ (NSDictionary *) verifyJWT:(NSString *) signature
```

Comment:
I can see that other `verify` methods here return `BOOL`  value (that make sense!) but this one is not. Maybe `parseJWT` should better reflect the purpose of it.

Reply:
@max-mironov I'll tried to make code on android and ios looks the same as possible, so that's why I call this function that way. If it is crucial, I can change it on both platforms, what do you think?

Reply:
yep, looks like this can be modified for both SDK's